modules/core/include/opencv2/core/mat.hpp

```diff
@@ -59,8 +59,12 @@ namespace cv
 //! @addtogroup core_basic
 //! @{
 
-enum { ACCESS_READ=1<<24, ACCESS_WRITE=1<<25,
-    ACCESS_RW=3<<24, ACCESS_MASK=ACCESS_RW, ACCESS_FAST=1<<26 };
+enum {
+    ACCESS_READ=1<<24,  //!< device read only, @sa Mat::getUMat
+    ACCESS_WRITE=1<<25, //!< device write only, @sa Mat::getUMat
+    ACCESS_RW=3<<24,    //!< device read and write, @sa Mat::getUMat
```

Comment:
This description doesn't work in some cases, like UMat::getMat() method and others (usage of this is hidden under "int" parameter).

This flag is requested access type / determines data flow / etc. But it is not linked to "device" directly.